<!DOCTYPE html>
<html>
<head>
    <title></title>
</head>
<body>

<script>

    try{
//        throw new  Error( "file not found " );
//        throw {'message':'file not fond!'};
//        throw 'file not fond';
//        throw {name:'error',message:'file not fond'}
//        throw 42;
//        throw true;
//        throw new  UserException( "file not found " );
        throw a;
    } catch(e){
        console.log(e);
    } finally{
        alert('最终一定执行');
    }

    function UserException(message) {
        this.message = message;
        this.name = "UserException";
    }

</script>

</body>
</html>